### DISCUSSION TITLE
Is it possible to display correct line numbers when using Eclipse Search panel to display References

### DISCUSSION THREAD

**Discussion by sdaimwood** [eclipse/lsp4e#784] [Q&A] [answered]
Hi there, I'm trying to display the results of a references query in the Search field of the eclipse workspace. The search display is currently displaying the `linenumber - 1` for the reference's line. I've debugged with the code and clearly it is displaying the Location object's exact value as the symbols are sent back from the language server and processed in `LSSearchQuery`. However, the location is zero-indexed, which doesn't match up with Eclipse's displayed line numbers. Curious if I'm missing a workaround or if this is something that needs to be fixed in LSP4E.

Due to my business organization's limits, we're using v0.20.7 and Eclipse 4.23, I tried to see if something like this issue has been fixed in subsequent releases but couldn't find anything definite. Didn't want to create an issue in case it would be a duplicate. 

See screenshots for where the references arrows are shown in the document and how the Search panel displays the `linenumber - 1` as it's represented in the language server. 

<img width="220" alt="image" src="https://github.com/eclipse/lsp4e/assets/93732205/c582ae78-78b8-4877-8c8e-bf39761744e2">

<img width="834" alt="Screenshot 2023-08-29 at 4 51 04 PM" src="https://github.com/eclipse/lsp4e/assets/93732205/4be740bf-41e1-4b9f-a48e-3819a248889e">

**Comment by sdaimwood** [comment]
Did some more debugging and found that if you modify the following lines:

https://github.com/eclipse/lsp4e/blob/9c92a321ea49efa1bc3dc746b469fbd52d6ccb0c/org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/references/LSSearchQuery.java#L131-L133

to simply add +1 when constructing the LineElement:

```java
 final var lineEntry = new LineElement(resource, document.getLineOfOffset(startOffset)+1, 
 		lineInformation.getOffset(), 
 		document.get(lineInformation.getOffset(), lineInformation.getLength())); 
```

that the display is correct, and the reference still goes to the correct location in the document. 

<img width="1001" alt="Screenshot 2023-08-29 at 5 09 07 PM" src="https://github.com/eclipse/lsp4e/assets/93732205/d2c181a6-44af-4cda-bf30-3b36c982bb33">

**Reply by mickaelistria** [reply, marked as answer]
Thanks for the investigation. Please submit a pull request for this patch.

**Reply by sdaimwood** [reply]
Will do, thank you for the confirmation 👍
